tests/unit/errors/test_common_functionality.py

```diff
@@ -14,104 +15,127 @@ def _strmatch_any_order(inputstr, prefix, midfixes, suffix, sep=", "):
     ]
 
 
-def test_raw_json_works(default_json_response):
-    err = GlobusAPIError(default_json_response.r)
-    assert err.raw_json == default_json_response.data
+def test_raw_json_works():
+    data = {
+        "code": "Json Error",
+        "errors": [
+            {
+                "message": "json error message",
+                "title": "json error message",
+            }
+        ],
+    }
 
+    err = construct_error(error_class=GlobusAPIError, body=data, http_status=400)
+    assert err.raw_json == data
 
-def test_raw_json_fail(default_text_response, malformed_response):
-    err = GlobusAPIError(default_text_response.r)
-    assert err.raw_json is None
 
-    err = GlobusAPIError(malformed_response.r)
+@pytest.mark.parametrize(
+    "body, response_headers",
+    (
+        ("text_data", {}),  # plain text
+        ("{", {"Content-Type": "application/json"}),  # malformed JSON
+    ),
+)
+def test_raw_json_none_on_nonjson_data(body, response_headers):
+    err = construct_error(
+        error_class=GlobusAPIError,
+        body=body,
+        response_headers=response_headers,
+        http_status=400,
+    )
     assert err.raw_json is None
 
 
-def test_text_property_works(default_json_response, default_text_response):
-    err = GlobusAPIError(default_json_response.r)
-    assert err.text == json.dumps(default_json_response.data)
-    err = GlobusAPIError(default_text_response.r)
-    assert err.text == default_text_response.data
+@pytest.mark.parametrize("body", ("text_data", {"foo": "bar"}))
```

Comment:
`body` can be bytes, too.

Reply:
Good catch! Let me add that.

Reply:
In my refactor, this actually folded into the suite of tests which are meant to exercise `construct_error`.
In this file, you'll see this trimmed back to a much simpler test which ensures that `GlobusAPIError.text` returns the string which was passed as the `body`.